lib/matplotlib/figure.py

```diff
@@ -1113,9 +1113,29 @@ def legend(self, *args, **kwargs):
             # extra_args = extra_args[1:]
             pass
         transform = kwargs.pop('bbox_transform', self.transSubfigure)
+        outside = None
+        if 'loc' in kwargs:
+            # need to protect this in case loc was passed in
+            # positionally...
+            loc = kwargs.pop('loc', None)
+            # handle outside legends:
+            if isinstance(loc, str):
+                if loc.split()[0] == 'outside':
+                    loc = loc.replace('outside ', '')
```

Comment:
```suggestion
                    loc = loc.split(' ', 1)[1]
```
Makes this permissive to multiple spaces. And does not silently accept strings containing multple 'outside' strings like 'outside left outside top'.

If we were Python 3.9+ you could also do `loc = loc.removeprefix('outside').lstrip()` but we're not yet there.